src/Features/Core/Portable/CodeStyle/AbstractCodeStyleProvider.Fixing.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    // This part contains all the logic for hooking up the CodeFixProvider to the CodeStyleProvider.
+    // All the code in this part is an implementation detail and is intentionally private so that
+    // subclasses cannot change anything.  All code relevant to subclasses relating to fixing is
+    // contained in AbstractCodeStyleProvider.cs
+
+    internal abstract partial class AbstractCodeStyleProvider<TOptionKind, TCodeStyleProvider>
+    {
+        private async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var diagnostic = context.Diagnostics[0];
```

Comment:
>[0] [](start = 48, length = 3)

What if empty?

Reply:
can't be.  that would be a violation of the contract around code fixes.  i.e. you only ever get called here with a diagnostic to fix.  @mavasani  to confirm.